main.c

```diff
@@ -693,12 +694,103 @@ _install_transaction (Pool         *pool,
     {
       Id p = installedq.elements[x];
       queue_pushunique (pile, p);
-      g_debug ("%*c - %s", indent, ' ', pool_solvid2str (pool, p));
+      const gchar *solvable = pool_solvid2str (pool, p);
+      g_debug ("%*c - %s", indent, ' ', solvable);
+      /* Non-modules are immediately resolved as resolved, since for RPM the
+       * result would not change if done again. However for modules we need to
+       * make sure we look at all combinations. */
+      if (!g_str_has_prefix (solvable, "module:"))
+        map_set (tested, p);
     }
 
   return TRUE;
 }
 
+/**
+ * mask_bare_rpms:
+ * @pool: initialized pool
+ *
+ * For each available modular package, find all bare RPMs with the same name,
+ * and mark them as not considered.
+ */
+static void
+mask_bare_rpms(Pool *pool)
+{
+  /* Get array of all existing modular packages. */
+  Id *modular_packages = pool_whatprovides_ptr (pool, pool_str2id (pool, MODPKG_PROV, 1));
+
+  /* Filter it to keep only available packages. A package that is not
+   * considered should not mask anything. */
+  g_auto(Queue) available_modular_pkgs;
+  queue_init(&available_modular_pkgs);
+  for (Id *pp = modular_packages; *pp; pp++)
+    {
+      if (map_tst (pool->considered, *pp))
+        {
+          queue_push2 (&available_modular_pkgs, SOLVER_SOLVABLE | SOLVER_INSTALL, *pp);
```

Comment:
no, this is wrong. Do selection, then substract unneeded solvables. don't create some overengineered stuff